internal/hmyapi/blockchain.go

```diff
@@ -63,6 +72,44 @@ func (s *PublicBlockChainAPI) GetBlockByHash(ctx context.Context, blockHash comm
 	return nil, err
 }
 
+// GenShardingStructure return sharding structure, given shard number and its patterns.
+func GenShardingStructure(shardNum, shardID uint32, httpPattern, wsPattern string) []map[string]interface{} {
```

Comment:
can you move this function to the schedule?

Reply:
Do you mean to introduce an interface and implement it for each type of network?

can we do it later as it's the blocker for beta?